### PR TITLE
feat(bigtable): Support copy backup in admin client

### PR DESCRIPTION
**Pull request by bhshkh**
This PR adds CopyBackup method to admin client using which existing backups can be copied to a same or different project.

### CODE REVIEW

**Review by gkevinzheng — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -132,6 +133,7 @@ project's service account.
 - `GCLOUD_TESTS_GOLANG_FIRESTORE_KEY`: The path to the JSON key file of the
 Firestore project's service account.
 - `GCLOUD_TESTS_API_KEY`: API key for using the Translate API created above.
+- `GCLOUD_TESTS_GOLANG_BIGTABLE_PROJECT_ID`: Developers Console project's ID (e.g. doorway-cliff-677) for the Bigtable project. Bigtable project can be same as Firestore project or any project other than the general project.
```
Comment (kolea2) [inline CONTRIBUTING.md:136]: maybe we should signal in the variable name this is a secondary project id (something like `..._SECONDARY_BIGTABLE_PROJECT_ID` for copy backup testing and it is optional?

File: bigtable/export_test.go [hunk 2]
```diff
@@ -54,6 +54,7 @@ func init() {
 	flag.StringVar(&c.AdminEndpoint, "it.admin-endpoint", "", "Admin api host and port")
 	flag.StringVar(&c.DataEndpoint, "it.data-endpoint", "", "Data api host and port")
 	flag.StringVar(&c.Project, "it.project", "", "Project to use for integration test")
+	flag.StringVar(&c.Project2, "it.project2", "", "Second project to use for integration test")
```
Comment (kolea2) [inline bigtable/export_test.go:57]: again would be nice to explain this is only for copy backup testing

File: bigtable/export_test.go [hunk 3]
```diff
@@ -110,6 +113,9 @@ func NewIntegrationEnv() (IntegrationEnv, error) {
 	if c.Project == "" {
 		c.Project = os.Getenv("GCLOUD_TESTS_GOLANG_PROJECT_ID")
 	}
+	if c.Project2 == "" {
```
Comment (kolea2) [inline bigtable/export_test.go:116]: if the user does not specify a second project, maybe we can skip or warn on the copy backup test that a second project was unspecified so it cannot run?

File: bigtable/admin.go [hunk 4]
```diff
@@ -1920,6 +1924,27 @@ func (ac *AdminClient) CreateBackup(ctx context.Context, table, cluster, backup
 	return longrunning.InternalNewOperation(ac.lroClient, op).Wait(ctx, &resp)
 }
 
+// CopyBackup copies the specified source backup with the user-provided expire time.
+func (ac *AdminClient) CopyBackup(ctx context.Context, destProject, destInstance, destCluster, destBackup,
```
Comment (kolea2) [inline bigtable/admin.go:1928]: following the flow of the java client (https://github.com/googleapis/java-bigtable/blob/main/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/BigtableTableAdminClient.java#L1371-L1373), the order seems to be source parameters, then destination parameters. WDYT, should we reorder or is there a reason to keep as is?

**Review by kolea2 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (bhshkh) [reply CONTRIBUTING.md]: Done

File: bigtable/export_test.go (hunk 2 above)
Comment (bhshkh) [reply bigtable/export_test.go]: Done

File: bigtable/export_test.go (hunk 3 above)
Comment (bhshkh) [reply bigtable/export_test.go:116]: Done

File: bigtable/admin.go (hunk 4 above)
Comment (bhshkh) [reply bigtable/admin.go]: Reordered

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -62,12 +62,13 @@ intend only to run integration tests on a single package.
...
 #### GCP Setup
 
-To run the integrations tests, creation and configuration of two projects in
+To run the integrations tests, creation and configuration of three projects in
 the Google Developers Console is required: one specifically for Firestore
-integration tests, and another for all other integration tests. We'll refer to
-these projects as "general project" and "Firestore project".
+integration tests, one specifically for Bigtable integration tests, and another 
+for all other integration tests. We'll refer to these projects as 
+"general project", "Bigtable project" and "Firestore project".
```
Comment (kolea2) [inline CONTRIBUTING.md:69]: maybe worth adding a note that if you're not interested in working on or testing Bigtable, you can skip the third project.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -62,12 +62,13 @@ intend only to run integration tests on a single package.
...
 
-After creating each project, you must [create a service account](https://developers.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount)
+After creating Firestore and general project, you must [create a service account](https://developers.google.com/identity/protocols/OAuth2ServiceAccount#creatinganaccount)
```
Comment (kolea2) [inline CONTRIBUTING.md:71]: I think leaving this as "each" is fine

File: bigtable/integration_test.go [hunk 7]
```diff
@@ -3002,6 +3002,196 @@ func TestIntegration_InstanceUpdate(t *testing.T) {
...
+	diffInstanceId := uid.NewSpace(prefixOfInstanceResources, &uid.Options{Short: true})
+	diffInstance := diffInstanceId.New()
+	diffCluster := testEnv.Config().Cluster + "-d"
+	conf := &InstanceConf{
+		InstanceId:   diffInstance,
+		ClusterId:    diffCluster,
+		DisplayName:  "different test sourceInstance",
+		Zone:         instanceToCreateZone2,
+		InstanceType: DEVELOPMENT,
+		Labels:       map[string]string{"test-label-key": "test-label-value"},
```
Comment (kolea2) [inline bigtable/integration_test.go:3015]: may be nice to give these labels a `-2` or `-diff` suffix to help keep track

File: bigtable/integration_test.go [hunk 8]
```diff
@@ -3002,6 +3002,196 @@ func TestIntegration_InstanceUpdate(t *testing.T) {
...
+		}
+		defer destIAdminClient2.DeleteInstance(ctx, destProj2Inst1)
+		testcases = append(testcases, testcase{
+			desc:         "Copy backup to different project",
+			destProject:  destProj2,
+			destInstance: destProj2Inst1,
+			destCluster:  destProj2Inst1Cl1,
+		})
+	} else {
+		t.Logf("Secondary project not set, skipping copy backup to different project testing")
```
Comment (kolea2) [inline bigtable/integration_test.go:3154]: not sure about the logging levels here, but a warn level may be nicer if available

**Review by kolea2 — APPROVED** [review]
a few small nits, but overall LGTM!

File: CONTRIBUTING.md (hunk 5 above)
Comment (bhshkh) [reply CONTRIBUTING.md:69]: Done

File: CONTRIBUTING.md (hunk 6 above)
Comment (bhshkh) [reply CONTRIBUTING.md]: Done

File: bigtable/integration_test.go (hunk 7 above)
Comment (bhshkh) [reply bigtable/integration_test.go]: Done

File: bigtable/integration_test.go (hunk 8 above)
Comment (bhshkh) [reply bigtable/integration_test.go]: There is no warning log level available. Added "WARNING" string to log message.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -62,10 +62,14 @@ intend only to run integration tests on a single package.
...
 #### GCP Setup
 
-To run the integrations tests, creation and configuration of two projects in
+To run the integrations tests, creation and configuration of three projects in
 the Google Developers Console is required: one specifically for Firestore
-integration tests, and another for all other integration tests. We'll refer to
-these projects as "general project" and "Firestore project".
+integration tests, one specifically for Bigtable integration tests, and another 
+for all other integration tests. We'll refer to these projects as 
+"general project", "Bigtable project" and "Firestore project".
```
Comment (quartzmo) [inline CONTRIBUTING.md:69]: The order of these names should be the same as the descriptions in the previous sentence.

File: internal/kokoro/continuous.sh [hunk 10]
```diff
@@ -30,6 +30,7 @@ export GOOGLE_APPLICATION_CREDENTIALS=$KOKORO_KEYSTORE_DIR/72523_go_integration_
 # doing when changing / removing the next line.
 
 export GCLOUD_TESTS_GOLANG_PROJECT_ID=dulcet-port-762
+export GCLOUD_TESTS_GOLANG_SECONDARY_BIGTABLE_PROJECT_ID=gcloud-golang-firestore-tests
```
Comment (quartzmo) [inline internal/kokoro/continuous.sh:33]: Is this correct? Is the bigtable project the same as the firestore project?

File: bigtable/admin.go [hunk 11]
```diff
@@ -1920,6 +1924,27 @@ func (ac *AdminClient) CreateBackup(ctx context.Context, table, cluster, backup
...
+		BackupId:     destBackup,
+		SourceBackup: sourceBackupPath,
+		ExpireTime:   timestamppb.New(expireTime),
+	}
+
+	op, err := ac.tClient.CopyBackup(ctx, req)
+	if err != nil {
+		return err
+	}
+	resp := btapb.Backup{}
```
Comment (quartzmo) [inline bigtable/admin.go:1944]: Why is the response empty?

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -132,6 +136,7 @@ project's service account.
 - `GCLOUD_TESTS_GOLANG_FIRESTORE_KEY`: The path to the JSON key file of the
 Firestore project's service account.
 - `GCLOUD_TESTS_API_KEY`: API key for using the Translate API created above.
+- `GCLOUD_TESTS_GOLANG_SECONDARY_BIGTABLE_PROJECT_ID`: Developers Console project's ID (e.g. doorway-cliff-677) for Bigtable optional secondary project. This can be same as Firestore project or any project other than the general project.
```
Comment (quartzmo) [inline CONTRIBUTING.md:139]: Why isn't an env var needed for a Bigtable API key, similar to the other projects?

Comment (bhshkh) [reply CONTRIBUTING.md:139]: Only Translate API is using API key for running tests.
None of the Bigtable tests use the API key for authentication.

File: bigtable/admin.go (hunk 11 above)
Comment (bhshkh) [reply bigtable/admin.go:1944]: The Wait call in next line populates the response in this variable

File: internal/kokoro/continuous.sh (hunk 10 above)
Comment (bhshkh) [reply internal/kokoro/continuous.sh:33]: Bigtable tests run on GCLOUD_TESTS_GOLANG_PROJECT_ID. There is only one test being added as part of this PR that requires a second Bigtable project (GCLOUD_TESTS_GOLANG_SECONDARY_BIGTABLE_PROJECT_ID) i.e. there is very minimal use of second Bigtable project. Setting up a complete new project for a single test does not make sense. So, Firestore project is being reused here.

File: CONTRIBUTING.md (hunk 9 above)
Comment (bhshkh) [reply CONTRIBUTING.md]: Oh, yes. Corrected!

File: CONTRIBUTING.md (hunk 12 above)
Comment (quartzmo) [reply CONTRIBUTING.md:139]: Sorry, I meant JSON key file.

File: bigtable/admin.go (hunk 11 above)
Comment (quartzmo) [reply bigtable/admin.go:1944]: sg, thanks!

File: internal/kokoro/continuous.sh (hunk 10 above)
Comment (quartzmo) [reply internal/kokoro/continuous.sh:33]: OK

**Review by quartzmo — APPROVED** [review]

File: CONTRIBUTING.md (hunk 12 above)
Comment (quartzmo) [reply CONTRIBUTING.md:139]: It looks to me like the Bigtable test might now depend on `GCLOUD_TESTS_GOLANG_FIRESTORE_KEY`, is that correct?

**Pull request merged**